lib/src/inline_parser.dart

```diff
@@ -453,15 +483,15 @@ class _DelimiterRun {
     } else {
       preceding = parser.source.substring(runStart - 1, runStart);
     }
-    precededByPunctuation = punctuation.contains(preceding);
+    precededByPunctuation = punctuation.hasMatch(preceding);
```

Comment:
If this turns out to be too slow consider using a set instead of a regexp for the lookup (since it is a single character, should be OK).

Reply:
Good call. I'm hoping a once-compiled RegExp produces a fast state machine that is essentially that Set. 😛